tests/conftest.py

```diff
@@ -92,8 +94,93 @@ def env():
         shutil.rmtree(fake_home, ignore_errors=True)
 
 
+@pytest.fixture()
+def webapp(active_config):
+    from dallinger.experiment_server import sockets
+    app = sockets.app
+    app.config['DEBUG'] = True
+    app.config['TESTING'] = True
+    client = app.test_client()
+    yield client
```

Comment:
You're not cleaning up the config changes here, is that safe?

We can make this clean up after itself through:

    app.config.extend({'DEBUG': True, 'TESTING': True})
    yield ...
    app.config.data.popleft()

I suspect it's fine as the config is resetup each time anyway, but it might cause issues if the config fixture was at a class level for some reason.

Reply:
@MatthewWilkes It seems like it would be better to have the config automatically reset after each _test_, maybe by simply changing the scope of `reset_config()` to `function` (which I think is the default). That way you don't need to worry about possible gaps based on what combination of fixtures you're using. It's a lot better now, but earlier on I was experiencing a lot of crazy test interdependencies where adding or moving a test class triggered failures elsewhere, because one set of tests was relying on cleanup performed by another. What do you think?

Reply:
(I was initially worried about doing a lot of config setting and wiping, but if I'm not mistaken it's all in-memory and seems more or less instantaneous.)

Reply:
Actually, doesn't this reset the config by virtue of using the `active_config` fixture? 

It turns out that the `config` in question here is a `Flask.config.Config` object, not our `Configuration` object, but your suggestion could still be implemented with `update()` and `del`.